petab_select/candidate_space.py

```diff
@@ -529,6 +602,628 @@ def iterate():
+
+    def update_after_calibration(
+        self,
+        history: Dict[str, Model],
+        local_history: Dict[str, Model],
+        criterion: Criterion,
+    ) -> None:
+        """See `CandidateSpace.update_after_calibration`."""
+        # In case we jumped to most distant in the last iteration,
+        # here we reset the jumped variable to False
+        self.jumped_to_most_distant = False
+
+        self.history = history
+
+        if not self.update_from_local_history(
+            local_history=local_history, criterion=criterion
+        ):
+            logging.info("Switching method")
+            self.switch_method()
+            self.switch_inner_candidate_space(history)
+            logging.info(
+                "Method switched to ", self.inner_candidate_space.method
+            )
+
+        self.method_history.append(self.method)
+        return self.jumped_to_most_distant
+
+    def update_from_local_history(
+        self, local_history: Dict[str, Model], criterion: Criterion
+    ) -> bool:
+        """Update the self.best_models with the latest local_history
+        and determine if there was a new best model. If so, return
+        True. False otherwise."""
+
+        found_new_best = False
+        for model_id in local_history:
+            if (
+                self.best_model_of_current_run == VIRTUAL_INITIAL_MODEL
+                or default_compare(
+                    self.best_model_of_current_run,
+                    local_history[model_id],
+                    criterion,
+                )
+            ):
+                found_new_best = True
+                self.best_model_of_current_run = local_history[model_id]
+
+            if len(self.best_models) < self.most_distant_max_number:
+                self.insert_model_into_best_models(
+                    model_to_insert=local_history[model_id],
+                    criterion=criterion,
+                )
+            elif default_compare(
+                self.best_models[self.most_distant_max_number - 1],
+                local_history[model_id],
+                criterion,
+            ):
+                self.insert_model_into_best_models(
+                    model_to_insert=local_history[model_id],
+                    criterion=criterion,
+                )
+
+        self.best_models = self.best_models[: self.most_distant_max_number]
+
+        # When we switch to LATERAL method, we will do only one iteration with this
+        # method. So if we do it succesfully (i.e. that we found_new_best), we still
+        # want to switch method. This is why we put found_new_best to False, so we go
+        # into the method switching pipeline
+        if (
+            found_new_best
+            and self.method == Method.LATERAL
+            and self.swap_only_once
+        ):
+            self.swap_done_successfully = True
+            found_new_best = False
+        return found_new_best
```

Comment:
It's unintuitive that `found_new_best` is `False` if the lateral method finds a new best model, could rephrase/rewrite.

Reply:
Yes, that's true. I've renamed it to go_into_switch_method. I think this is much better.